### ISSUE TITLE
Performance Issues: Lag and Crashes – Request for Fix

### ISSUE THREAD

**Issue by mikropsoft** [getumbrel/umbrel#1858]
Since I've been using Umbrel on the RPi5, the performance has been terrible and truly problematic. Even the slightest task or downloading 2-3 apps causes the panel to start running extremely slowly, and sometimes the page even crashes. I would also like to point out that the RAM usage is not even over half, and the CPU usage does not consistently go over 60%. Despite these conditions, these issues still occur. Of course, since the monitoring panel also freezes, it becomes impossible for me to examine this further, but there is no significant strain on the device.

**Comment by mayankchhabra** [comment]
Hey @mikropsoft - that doesn't sound normal, I agree. I'm wondering if it's a browser issue vs. performance issue on the device. Would you be able to share the specs of the device you're using to access your Umbrel? Also, are you able to recreate the laggy UI on other devices, such as your phone?

**Comment by mikropsoft** [comment]
> Hey @mikropsoft - that doesn't sound normal, I agree. I'm wondering if it's a browser issue vs. performance issue on the device. Would you be able to share the specs of the device you're using to access your Umbrel? Also, are you able to recreate the laggy UI on other devices, such as your phone?

No, my devices are powerful enough. For example, I uploaded and confirmed a file in the Stirling PDF application to learn the PDF information from any setting. After that, the CPU usage doesn't even increase, but the page freezes. Then, when I try to access the Umbrel page, it crashes (this is just an example). As a result, it is not a problem with my devices. Overall, the interface is already overloaded.

**Comment by mikropsoft** [comment]
By the way, it's not an issue with storage either; my USB device is fast enough.

**Comment by mikropsoft** [comment]
Will it be taken into consideration @mayankchhabra

**Comment by nmfretz** [comment]
Hey @mikropsoft, would you be able to give us a bit more info on your specific device and browser you are accessing your Umbrel from? As @mayankchhabra mentioned this doesn't sound normal, and it may be something like a browser issue (e.g., a browser compatibility issue for rendering certain elements on the page) and not related to how powerful your devices are. This sounds like your browser is crashing when trying to render certain elements on the page or when making too many requests, not that your Raspberry Pi 5 and umbrelOS itself is crashing.

Could you please let us know:
1) What device you are using when accessing umbrel.local
2) What browser you are using when accessing umbrel.local
3) Do you experience the same issues when accessing umbrel.local from a different device or browser? For example, if the issue occurs when you are on your Windows PC using Firefox, does it also occur from your mobile phone using Chrome?
4) Posting a video recording here, as you suggest, would be great 🙏

**Comment by mikropsoft** [comment]
> Hey @mikropsoft, would you be able to give us a bit more info on your specific device and browser you are accessing your Umbrel from? As @mayankchhabra mentioned this doesn't sound normal, and it may be something like a browser issue (e.g., a browser compatibility issue for rendering certain elements on the page) and not related to how powerful your devices are. This sounds like your browser is crashing when trying to render certain elements on the page or when making too many requests, not that your Raspberry Pi 5 and umbrelOS itself is crashing.
> 
> Could you please let us know:
> 
>     1. What device you are using when accessing umbrel.local
> 
>     2. What browser you are using when accessing umbrel.local
> 
>     3. Do you experience the same issues when accessing umbrel.local from a different device or browser? For example, if the issue occurs when you are on your Windows PC using Firefox, does it also occur from your mobile phone using Chrome?
> 
>     4. Posting a video recording here, as you suggest, would be great 🙏
> 
> 
> This will help us better understand the issue and potentially identify if it’s a browser compatibility problem or something else.

1: I use a laptop and a phone connected via LAN when accessing Umbrel.

2: I use Firefox. Previously, I used Brave (both on PC and mobile). The issue is definitely not related to the browser.

3: As mentioned, I previously experienced the same issue on both devices while using Brave. Now, I am using Firefox, and there is no difference.

Recording a video will take some effort, and I will need to make a detailed video since the issue doesn’t occur consistently; it happens rarely. For example, I encounter an error when deleting any photo through Immich, which then causes Umbrel to crash.

I will inform you once I complete the video recording.

**Comment by mikropsoft** [comment]
@nmfretz https://drive.google.com/file/d/1n_dRhsQCNXdxQyrMTkJl6kLmOb_y_uzj/view?usp=drive_link

I acted a bit amateurishly while trying to demonstrate the issue. However, in the end, I was able to show the problem I mentioned. Since this is a rare occurrence, I had to try a bit too hard in the video. Sometimes it would crash even when I was just trying to upload a photo on immich. Additionally, towards the end of the video, I conducted the test using a different browser as well. Thank you.
